src/App.tsx

```diff
@@ -15,47 +16,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods
+        .sort((firstGood, secondGood) => (
+          firstGood.localeCompare(secondGood)
+        ));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((firstGood, secondGood) => (
+        firstGood.length - secondGood.length));
+      break;
+    case SortType.NONE:
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+  const sortByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverseGoods = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
```

Comment:
```suggestion
  const handleResetGoods= () => {
    setIsReversed(false);
    setSortType(SortType.NONE);
  };
```
![image](https://user-images.githubusercontent.com/91411692/234325320-4e970dd6-a1ae-41dd-a22a-89fc5de2cd2c.png)